src/main/kotlin/org/rust/cargo/toolchain/RsToolchainBase.kt

```diff
@@ -141,9 +141,9 @@ abstract class RsToolchainBase(val location: Path) {
 
             return RsToolchainFlavor.getApplicableFlavors()
                 .asSequence()
-                .flatMap { it.suggestHomePaths() }
+                .flatMap { it.suggestProjectPaths(projectDir) + it.suggestHomePaths() }
                 .mapNotNull { RsToolchainProvider.getToolchain(it.toAbsolutePath()) }
-                .firstOrNull()
+                .let { toolchains -> toolchains.firstOrNull { "bazel" in it.presentableLocation } ?: toolchains.firstOrNull() }
```

Comment:
If the user has `rustup` installed globally but we're in a Bazel project, prefer using the `rustc` and `cargo` installed by Bazel (note that Bazel allows you to specify a `rustc` version per-project)

This method of checking whether we're looking at a Bazel-installed Rust toolchain is slightly fragile, but unlikely to break unless the user has `bazel` in their name.